### PR TITLE
refactor: use `ExecutorParams::info` in stream scan executors

### PR DESCRIPTION
**Pull request by stdrc**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This is part of the executor info consistency work https://github.com/risingwavelabs/risingwave/issues/13292.

Summary:

- Directly use `ExecutorParams::info` added in #13342 instead of deriving `schema` in `ChainExecutorBuilder`.
- Clarify the naming of `table schema` and `schema`, `table pk` and executor `pk_indices` in `ChainExecutorBuilder::new_boxed_executor`.
- Rename `ChainExecutorBuilder` to `StreamScanExecutorBuilder`, which seems to be forgotten in #13242.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
LGTM

**Review by st1page — APPROVED** [review]

**Review by chenzl25 — APPROVED** [review]

File: src/stream/src/from_proto/stream_scan.rs [hunk 1]
```diff
@@ -104,16 +71,16 @@ impl ExecutorBuilder for ChainExecutorBuilder {
                     .map(|(k, v)| (k.clone(), v.clone()))
                     .collect();
                 let table_type = CdcTableType::from_properties(&properties);
-                let table_reader =
-                    table_type.create_table_reader(properties.clone(), schema.clone())?;
+                let table_reader = table_type
+                    .create_table_reader(properties.clone(), params.info.schema.clone())?;
```
Comment (StrikeW) [inline src/stream/src/from_proto/stream_scan.rs:75]: May I know where we derive the `info.schema`?

Comment (stdrc) [reply src/stream/src/from_proto/stream_scan.rs]: It's from https://github.com/risingwavelabs/risingwave/blob/a288378387699a09f9e7b50b193f6e2bc8cc8eb4/src/frontend/src/optimizer/plan_node/stream_table_scan.rs#L350-L351

The `self.schema()` is derived at https://github.com/risingwavelabs/risingwave/blob/a288378387699a09f9e7b50b193f6e2bc8cc8eb4/src/frontend/src/optimizer/plan_node/generic/scan.rs#L374-L385

Comment (stdrc) [reply src/stream/src/from_proto/stream_scan.rs]: I adjusted the cdc backfill executor construction logic, should be more reasonable now. PTAL again🥰

**Pull request merged**